shared/src/components/BadgeAttachment.tsx

```diff
@@ -1,38 +1,68 @@
 import * as React from 'react'
 import isAbsoluteUrl from 'is-absolute-url'
+import InformationIcon from 'mdi-react/InfoCircleOutlineIcon'
+import WarningIcon from 'mdi-react/AlertCircleOutlineIcon'
+import ErrorIcon from 'mdi-react/AlertDecagramOutlineIcon'
 import { BadgeAttachmentRenderOptions } from 'sourcegraph'
 import { badgeAttachmentStyleForTheme } from '../api/client/services/decoration'
 import { LinkOrSpan } from './LinkOrSpan'
 import { isEncodedImage } from '../util/icon'
+import { MdiReactIconComponentType } from 'mdi-react'
+import classNames from 'classnames'
 
-export const BadgeAttachment: React.FunctionComponent<{
-    attachment: BadgeAttachmentRenderOptions
-    isLightTheme: boolean
-}> = ({ attachment, isLightTheme }) => {
-    const style = badgeAttachmentStyleForTheme(attachment, isLightTheme)
+const chooseIconComponent = (icon: BadgeAttachmentRenderOptions['kind']): MdiReactIconComponentType => {
+    switch (icon) {
+        case 'info':
+            return InformationIcon
+        case 'warning':
+            return WarningIcon
+        case 'error':
+            return ErrorIcon
+    }
+}
+
+const isPredefinedIcon = (badge: BadgeAttachmentRenderOptions): boolean => 'kind' in badge
+
+const renderIcon = (badge: BadgeAttachmentRenderOptions, isLightTheme: boolean): JSX.Element | null => {
+    if (isPredefinedIcon(badge)) {
+        const Icon = chooseIconComponent(badge.kind)
+        return <Icon className={classNames('icon-inline', 'badge-decoration-attachment__icon-svg')} />
```

Comment:
Nit: Since these are not dynamic, you can just pass them as a string.
```suggestion
        return <Icon className="icon-inline badge-decoration-attachment__icon-svg" />
```

Reply:
That is a good point. I felt it is better to copy/pasting and navigating individually (I have a shortcut that selects a string content).
In terms of perf there is not much impact here.
1. It rerenders infrequently and there are not many instances of it.
2. A single jsx element creation <button onClick={...}/> dwarfs any perf concerns for classNames (in my experience)

Reply:
My concern here was not performance, just readability. The JSX binding, function call, individual quoting and commas all add a lot more syntax noise, and using just the string is consistent with the rest of our codebase.

I think generally there shouldn't be a need to copy-paste classes, because they are either  specific to the element, or Bootstrap utility classes with autocompletion. But VS Code also has "Expand selection" which just selects the current word :)

Reply:
fair enough